pinecone/test_suite.go

```diff
@@ -87,7 +87,9 @@ func (ts *IntegrationTests) TearDownSuite() {
 	_, err = WaitUntilIndexReady(ts, ctx)
 	require.NoError(ts.T(), err)
 	err = ts.client.DeleteIndex(ctx, ts.idxName)
-	require.NoError(ts.T(), err)
+	if err != nil {
```

Comment:
Do you need to retry here? I know sometimes w/configure index calls and the like, we have to retry in the TS client

Reply:
Added a small retry mechanism here for cleanup.